src/ralph/backends/storage/swift.py

```diff
@@ -56,50 +46,54 @@ def __init__(
     ):
         """Setup options for the SwiftService"""
 
-        self.options = {
-            "os_auth_url": os_auth_url,
-            "os_identity_api_version": os_identity_api_version,
-            "os_user_domain_name": os_user_domain_name,
-            "os_project_domain_name": os_project_domain_name,
-            "os_tenant_id": os_tenant_id,
-            "os_tenant_name": os_tenant_name,
-            "os_username": os_username,
-            "os_password": os_password,
-            "os_region_name": os_region_name,
-            "os_storage_url": os_storage_url,
-        }
-        self._check_options()
-        self.container = os_storage_url.split("/")[-1]
-        self.options["os_storage_url"] = os_storage_url.replace(
-            f"/{self.container}", ""
-        )
-
-    def _check_options(self):
-        """Verify that all options are not None nor empty"""
+        self.os_tenant_id = os_tenant_id
+        self.os_tenant_name = os_tenant_name
+        self.os_username = os_username
+        self.os_password = os_password
+        self.os_region_name = os_region_name
+        self.os_user_domain_name = os_user_domain_name
+        self.os_project_domain_name = os_project_domain_name
+        self.os_auth_url = os_auth_url
+        self.os_identity_api_version = os_identity_api_version
+        self.container = urlparse(os_storage_url).path.rpartition("/")[-1]
+        self.os_storage_url = os_storage_url.replace(f"/{self.container}", "")
 
-        for option_name in self.options:
-            option = self.options[option_name]
-            # pylint: disable=protected-access
-            if option is not None and not isinstance(option, inspect._empty):
-                continue
-            msg = "SwiftStorage backend instance requires the %s option to be set"
-            logger.error(msg, option_name)
-            raise BackendParameterException(msg % option_name)
+        with SwiftService(self.options) as swift:
+            stats = swift.stat()
+            if not stats["success"]:
+                msg = "SwiftStorage backend unable to connect: %s"
```

Comment:
The class/method is in the logger context (depending on the log format, you can choose to display it or not), we don't need to repeat it.

```suggestion
                msg = "Unable to connect to the requested container: %s"
```

Reply:
Indeed, Thanks)